pkg/operator/deploy/deploy.go

```diff
@@ -83,49 +83,74 @@ func New(log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, arocli
 	}, nil
 }
 
-func (o *operator) staticResources() ([]kruntime.Object, error) {
-	results := []kruntime.Object{}
-	for _, fileBytes := range utilembed.ReadDirRecursive(embeddedFiles, "staticresources") {
-		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(fileBytes, nil, nil)
-		if err != nil {
-			return nil, err
-		}
+type deployData struct {
+	Image              string
+	Version            string
+	GitCommit          string
+	IsLocalDevelopment bool
+	HasVersion         bool
+}
 
-		// set the image for the deployments
-		if d, ok := obj.(*appsv1.Deployment); ok {
-			if d.Labels == nil {
-				d.Labels = map[string]string{}
-			}
-			var image string
-
-			if o.oc.Properties.OperatorVersion != "" {
-				image = fmt.Sprintf("%s/aro:%s", o.env.ACRDomain(), o.oc.Properties.OperatorVersion)
-				d.Labels["version"] = o.oc.Properties.OperatorVersion
-			} else {
-				image = o.env.AROOperatorImage()
-				d.Labels["version"] = version.GitCommit
-			}
+func templateManifests(data deployData) ([][]byte, error) {
```

Comment:
IMHO the function can be simplified a bit by using an aux function to execute the templates (I think it reads a bit better and narrows the scope of some variables like "result". I also propose a more specific name rather than "result"). 

Important question🧐: why are you using "buff.Reset()" if buff is declared inside the most nested "for" (so it is instantiated in each iteration)? If you want to reuse "buff" shouldn't you declare it outside that "for" loop? Is that a minor error or I am missing something (this is more probable  🙂)? 

I propose something like this (also a minor change: I have reordered the declaration of templates to be in the same order as they appear in the templatesArray and extract the common string to a constant).

```go
func executeTemplates(templatesArray []*template.Template, data deployData) ([][]byte, error) {
	templateExecutionResult := make([][]byte, 0)

	for _, templates := range templatesArray {
		for _, v := range templates.Templates() {
			buff := &bytes.Buffer{}
			if err := v.Execute(buff, data); err != nil {
				return nil, err
			}
			templateExecutionResult = append(templateExecutionResult, buff.Bytes())
			buff.Reset()
		}
	}

	return templateExecutionResult, nil
}

func templateManifests(data deployData) ([][]byte, error) {
	const ResourcesFolder = "staticresources"

	templatesMaster, err := template.ParseFS(embeddedFiles, ResourcesFolder+"/master/*")
	if err != nil {
		return nil, err
	}

	templatesRoot, err := template.ParseFS(embeddedFiles, ResourcesFolder+"/*.yaml")
	if err != nil {
		return nil, err
	}

	templatesWorker, err := template.ParseFS(embeddedFiles, ResourcesFolder+"/worker/*")
	if err != nil {
		return nil, err
	}

	templatesArray := []*template.Template{templatesMaster, templatesRoot, templatesWorker}

	result, err := executeTemplates(templatesArray, data)
	if err != nil {
		return nil, err
	}

	return result, nil
}
```

Reply:
I think it is fine to keep it this way, there is no complexity in the reading of the templates

Reply:
for the reset, I just prefer to reset buffers once used. It is just a habit so I don't forget to reset those.

Reply:
Maybe I am wrong but speaking about the "reset", I think it is not just unnecessary but also incorrect as the variable buff is re-instantiated in every iteration of the for loop and the reset is having no effect -> so you are basically resetting an object that will be deleted in the next "line".

For the part of the template, it was just a proposal to apply the [SRP](https://levelup.gitconnected.com/practical-solid-in-golang-single-responsibility-principle-20afb8643483) because if the way the templates are executed changes, you just want to touch the function responsible for that, not a function that it's also responsible to obtain the templates-> mixed responsibilities. 

SRP improves the codebase health in the long run.